x-pack/server/lib/create_router/index.d.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Request, ResponseObject, ResponseToolkit } from 'hapi';
+
+type CallWithRequest = (request: any, action: string, params: any) => any;
```

Comment:
I believe we can type `request` per my comment above.

Reply:
I think the legacy ES call stuff is all typed now: https://github.com/elastic/kibana/blob/master/src/legacy/core_plugins/elasticsearch/index.d.ts